pkg/pq/priority_queue.go

```diff
@@ -51,16 +51,34 @@ func (pq *PriorityQueue) Len() int {
 }
 
 func (pq *PriorityQueue) Release(value Value) {
-	queue := &internalQueue{}
-	heap.Init(queue)
+	idx := pq.findIdx(value)
 
+	if idx == -1 {
+		return
+	}
+
+	lastIdx := (*pq.queue).Len() - 1
+	lastItem := (*pq.queue)[lastIdx]
+
+	lastItem.index = idx
+	(*pq.queue)[idx] = lastItem
+	heap.Fix(pq.queue, idx)
+
+	(*pq.queue)[lastIdx] = nil
+	(*pq.queue) = (*pq.queue)[:lastIdx]
+
+	if (*pq.queue).Len() == 0 {
+		pq.queue = &internalQueue{}
+	}
+}
+
+func (pq *PriorityQueue) findIdx(value Value) int {
 	for _, item := range *pq.queue {
-		if item.value != value {
-			heap.Push(queue, item)
+		if item.value == value {
+			return item.index
```

Comment:
I am not sure how to use the index of the Item.

This is because the item in the current code is not entered as an argument value, and the value is entered.